core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/environment/network/EnvironmentBaseNetworkConverter.java

```diff
@@ -66,5 +49,43 @@ public Network convertToLegacyNetwork(EnvironmentNetworkResponse source, String
         return result;
     }
 
+    private Map<String, Object> buildAttributes(EnvironmentNetworkResponse source, String availabilityZone) {
+        Map<String, Object> attributes = new HashMap<>();
+        CloudSubnet cloudSubnet = getCloudSubnet(source, availabilityZone);
+        LOGGER.debug("Chosen subnet: {}", cloudSubnet);
+
+        attributes.put(SUBNET_ID, cloudSubnet.getId());
+        attributes.put(CLOUD_PLATFORM, getCloudPlatform().name());
+        attributes.putAll(getAttributesForLegacyNetwork(source));
+        if (PublicEndpointAccessGateway.ENABLED.equals(source.getPublicEndpointAccessGateway())) {
+            attachEndpointGatewaySubnet(source, attributes, cloudSubnet);
+        }
+        return attributes;
+    }
+
+    /**
+     * Attach the subnet ID in which to place a public endpoint gateway.
+     *
+     * The default implementation should be sufficient for GCP and AWS.
+     *
+     * For Azure, we do not need to attach the endpoint gateway to a public subnet, so this method must be overrridable.
+     * @param source contains source information to retrieve subnets from
+     * @param attributes a Map which we put the gateway subnet ID in
+     */
+    protected void attachEndpointGatewaySubnet(EnvironmentNetworkResponse source, Map<String, Object> attributes, CloudSubnet cloudSubnet) {
+        Optional<CloudSubnet> endpointGatewaySubnet = subnetSelector.chooseSubnetForEndpointGateway(source, cloudSubnet.getId());
+        if (endpointGatewaySubnet.isPresent()) {
+            attributes.put(ENDPOINT_GATEWAY_SUBNET_ID, endpointGatewaySubnet.get().getId());
+        } else {
+            throw new BadRequestException("Could not find public subnet in availability zone: " + cloudSubnet.getAvailabilityZone());
+        }
```

Comment:
I think you could also change this logic to use the `orElseThrow()` like you did in `getCloudSubnet`.